document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -263,8 +268,21 @@ public BulkUpdateResult bulkUpdateSubDocs(Map<Key, Map<String, Document>> docume
   }
 
   @Override
-  public BulkUpdateResult bulkOperationOnArrayValue(BulkArrayValueUpdateRequest request) {
-    throw new UnsupportedOperationException();
+  public BulkUpdateResult bulkOperationOnArrayValue(BulkArrayValueUpdateRequest request)
+      throws Exception {
+    Operation operation = request.getOperation();
+    switch (operation) {
+        // todo: All of these implementations do a GET -> in memory update -> UPSERT which is not
```

Comment:
Let's add this as a method-level comment, explaining the approach.

/**The approach was taken for in-memory join as there is no direct support for set operation on JSON array. If, needed, this can be optimized or implemented using jsonb_set or jsonb_insert*/